src/ProjectTemplates/test/BaselineTest.cs

```diff
@@ -116,37 +103,58 @@ public async Task Template_Produces_The_Right_Set_Of_FilesAsync(string arguments
             }
         }
 
-        private string SanitizeArgs(string arguments)
-        {
-            var text = TemplateNameRegex.Match(arguments)
-                .Groups.TryGetValue("template", out var template) ? template.Value : "";
+        private static ConcurrentDictionary<string, object> _projectKeys = new();
 
-            text += AuthenticationOptionRegex.Match(arguments)
-                .Groups.TryGetValue("auth", out var auth) ? auth.Value : "";
+        private string CreateProjectKey(string arguments)
```

Comment:
Only minor question I had was to confirm that we don't care about ordering variation... since "dotnet new blazorwasm -ho --auth Individual" won't be considered the same as "dotnet new blazorwasm --auth Individual -ho", probably not an issue since our tests hopefully are always consistent

Reply:
That's a good question as this routine would now produce a unique key. I think the "worst" that would happen in that case is we end up with a duplicate test, rather than the worst case now, being tests that should be unique, aren't, and the tests fail with somewhat obscure errors.

Reply:
I can add a sort to make it a non-issue.

Reply:
@HaoK see https://github.com/dotnet/aspnetcore/pull/36068/commits/3314c7f0d01991395a824642b387206fcbe1d409